pkg/reconciler/v1alpha1/pipelinerun/pipelinerun_test.go

```diff
@@ -278,6 +282,43 @@ func TestReconcile(t *testing.T) {
 	if _, exists := reconciledRun.Status.TaskRuns["test-pipeline-run-success-unit-test-1"]; exists == false {
 		t.Errorf("Expected PipelineRun status to include TaskRun status")
 	}
+
+	if err := clients.Pipeline.Pipeline().TaskRuns("foo").Delete("test-pipeline-run-success-unit-test-1", &metav1.DeleteOptions{GracePeriodSeconds:new(int64)}); err != nil {
+		t.Fatalf("Failed to delete TaskRun: %s", err)
+	}
+
+	reconciledRun.Status.SetCondition(&duckv1alpha1.Condition{
+		Type:    duckv1alpha1.ConditionSucceeded,
+		Status:  corev1.ConditionTrue,
+		Reason:  resources.ReasonSucceeded,
+		Message: "All Tasks have completed executing",
+	})
+
+	if _, err := clients.Pipeline.Pipeline().PipelineRuns("foo").Update(reconciledRun); err != nil {
+		t.Fatalf("Error updating PipelineRun: %s", err)
+	}
+
+	informers.PipelineRun.Informer().GetIndexer().Add(reconciledRun)
+
+	if err := c.Reconciler.Reconcile(context.Background(), "foo/test-pipeline-run-success"); err != nil {
+		t.Errorf("Did not expect to see error when reconciling valid Pipeline but saw %s", err)
+	}
+
+	// Check that the PipelineRun was not changed by reconciliation
+	twiceReconciledRun, err := clients.Pipeline.Pipeline().PipelineRuns("foo").Get("test-pipeline-run-success", metav1.GetOptions{})
```

Comment:
Ah, I assumed the existing object wouldn't be updated, but I honestly never get when anything is a reference rather than a value in the first place. =)

Reply:
haha yeah i get that too :D